docs/system_overview/process_model.md

```diff
@@ -1,29 +1,36 @@
+This process immediately creates two other processes:
 
-Agent processes are listed with the IP address of the connecting client.
+- **Agent Factory**: Handles all incoming API requests
+- **Delay Server** (optional): Manages delayed rule execution
 
-    /usr/sbin/irodsServer
-    ├─ irodsDelayServer
-    └─ irodsAgentFactory
-       ├─ irodsAgent
-       ├─ irodsAgent
-       ├─ irodsAgent
-       └─ irodsAgent
+## How requests are handled
 
-| iRODS Process         | Expected Duration        | Expected Resident Memory Usage  |
-| --------------------- | ------------------------ | ------------------------------- |
-| irodsServer           | long-running             | 126M                            |
-| irodsDelayServer      | long-running             | 17M                             |
-| irodsAgentFactory     | long-running             | 5M                              |
-| irodsAgent            | single client connection | 20M to 100M                     |
+The **Agent Factory** listens for new client connections. For each new connection, it forks a separate **Agent** process. Each Agent handles one client request, completes the work as quickly as possible, and then waits for the next request. Agents will continue to service requests until the client disconnects.
 
-iRODS uses a dynamic linking model, rather than a static linking model, which adds significant overhead for short running processes (e.g. `ils`).  In order to mitigate this overhead, the irodsServer no longer calls `fork()` and `exec()` as a separate executable (pre-4.2).  In addition, since calling `fork()` within a threaded environment creates undefined behavior, the irodsServer calls `fork()` to spawn the Agent Factory process early, before any threading begins and any configuration is determined.
+## The Delay Server
 
-![iRODS Process Model Diagram](../images/process_model_diagram.jpg)
+The **Delay Server** is designed to remain idle most of the time. Every 30 seconds _(by default)_, it wakes up and spawns an Agent to check for any delayed rules that need to run. You can adjust this interval using the `delay_server_sleep_time_in_seconds` property in server_config.json. See [Configuration](../system_overview/configuration.md#etcirodsserver_configjson) for more information about this property.
 
+## Example Process Tree
 
+    irodsServer
+      ├─irodsAgent
+      │   ├─irodsAgent
+      │   ├─irodsAgent
+      │   └─irodsAgent
+      └─irodsDelayServer
 
+| iRODS Process         | Expected Duration        | Expected Memory Usage           |
+| --------------------- | ------------------------ | ------------------------------- |
+| irodsServer           | Long-running             | ~34 MB                          |
+| irodsDelayServer      | Long-running             | ~33 MB                          |
+| irodsAgent (factory)  | Long-running             | ~92 MB                          |
+| irodsAgent            | Depends on client        | ~92 MB (minimum)                |
```

Comment:
While the use of "Example" in the title isn't wrong, it feels out of place to me.

Try adding a statement which explains that the following structure is an example of what the process tree for iRODS could look like.

Could be helpful to include words about the table too.

Reply:
Example ....  could be replaced with ...

Standard
Representative
Common
Usual

and consider reordering the table so delayserver is last... like in the text tree

Reply:
and yes, a could sentences about the table would be good

Reply:
"Expected"?

Reply:
Updated.